File: 24. Swap Nodes in Pairs/swapPairs.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode swapPairs(ListNode head) {
        if (head == null || head.next == null) return head;
        ListNode former = head;
        ListNode latter = former.next;
        while (former != null && latter != null){
            int a = former.val;
            former.val = latter.val;
            latter.val = a;
            former = former.next.next;
            if (latter.next != null){
                latter = latter.next.next;
            }
            else{
                return head;
            }
        }
        return head;
    }
}

File: 24. Swap Nodes in Pairs/swapPairs.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def swapPairs(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        result = pre = ListNode(0)
        pre.next = head
        while pre.next and pre.next.next:
            #    pre -> first -> second -> second.next
            # => pre -> second -> first -> second.next
            first = pre.next
            second = first.next
            pre.next, second.next, first.next = second, first, second.next
            pre = first
        return result.next